test/dart/devtools_integration/delivery_test.dart

```diff
@@ -0,0 +1,45 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:leak_tracker/devtools_integration.dart';
+import 'package:leak_tracker/src/devtools_integration/_envelopes.dart';
+import 'package:test/test.dart';
+
+import '../../test_infra/data/messages.dart';
+
+void main() {
+  setUp(() {
+    verifyTestsCoverAllEnvelopes();
+  });
+
+  for (final message in messages) {
+    test('delivery classes serialize message', () {
+      final envelope = envelopeByType(message.runtimeType);
+
+      switch (envelope.channel) {
+        case Channel.requestToApp:
```

Comment:
Doing a switch statement in unit tests decreases readability.
It might be more readable to make separate tests  where you grab the types you want, then operate on those types.